packages/frint-props/README.md

```diff
@@ -297,6 +298,29 @@ const props$ = withState('counter', 'setCounter', 0)(app);
 
 <!-- -->
 
+## withObservable
+
+> withObservable(source$, ...mappers)
+
+### Arguments
+
+1. `source$` (`Observable` OR function returning `Observable`)
```

Comment:
Do we really need to support functions here? Isn't it overcomplicating and making API too broad while making first step?

Reply:
there is a reason for supporting functions here, mainly to give access to the `app` instance.

take a look at this example:

```js
import { of } from 'rxjs/observable/of';

compose(
  withObservable(function (app) {
    return of({ name: app.getName() });
  })
)(app);
```

The example above only returns the App's name. But imaging accessing an `http` provider, that requests the server and then returns its response which will me ultimately made available in the streamed props object.

Functions would be very handy there. Besides, it will also support use cases when the props passed from parent component needs to be processed:

```js
compose(
  withObservable(function (app, parentProps$) {
    return Observable;
  })
)(app, parentProps$);
```

The function signature will follow what `observe` higher-order component receives as an argument in `frint-react`.

Reply:
Got it. Thanks for explanation.